build-recipe-preinstallimage

```diff
@@ -57,10 +57,21 @@ recipe_build_preinstallimage() {
       esac
       TOPDIRS="$TOPDIRS $DIR"
     done
-    if ! $TAR -czf .preinstallimage.$$.tar.gz --one-file-system $TOPDIRS ; then
+    if ! $TAR -cf .preinstallimage.$$.tar --one-file-system $TOPDIRS ; then
 	cleanup_and_exit 1
     fi
     echo "image created."
+    local compress=gzip
+    local ext=gz
+    preinstallimagetype=$(queryconfig --dist "$BUILD_DIST" --configdir "$CONFIG_DIR" --archpath "$BUILD_ARCH" buildflags preinstallimagetype)
+    if test "$preinstallimagetype" = "zstd" -a -n "$(type -p zstd)"; then
+        compress="zstd -19 -T0"
+        ext=zst
+    else
+        echo "Using legacy gzip preinstall images."
```

Comment:
I don't like the word "legacy". Just do "Using '$compress' for image compression".

Reply:
done